R/edgeThresh.R

```diff
@@ -0,0 +1,284 @@
+# A value for δ is chosen such that random networks will not result in large subnetworks.
+
+#' Define edge-removal threshold parameter δ for each target subnet-order, from
+#' random permutations of the input EGG object
+#'
+#' \code{THRESH} creates N permutations of the input graph EGG by randomly
+#' choosing 2 edges and swapping their target endpoints (i.e. B in the
+#' edge A → B). Finds the value δ (edge weight) for each random graph such that
+#' the maximum size of any component in the graph is Lmax, and returns the
+#' median of N δ values.
+#'
+#' @section Swapping Edges:
+#'     Bidirectional edges must only be swapped with each other, therefore the
+#'     swapping operation will first be applied to bidirectional edges,
+#'     Q * |E_b| times, where E_b is a bidirectional edge. Since all 4 incident
+#'     nodes in 2 bidirectional edges are receiving endpoints, one node will be
+#'     randomly selected from each edge, and a new edge is formed between the
+#'     selected as well as the remaining vertices.
+#'     Then Q * |E_u| unidirectional edge swaps take place.
+#'     At each unidirectional edge swap, only the receiving endpoints are
+#'     swapped e.g. (A → B, C → D) to (A → D, C → B).
+#'     In both cases, a swap should NOT result in:
+#'     Multiple edges: Possibly result of a bidirectional edge swap.
+#'     Self-edges: e.g. (A ↔ B, C ↔ A) to (A ↔ A, B ↔ C)
+#'     Disconnected components: Considering (A → B, C → D), if A and B are
```

Comment:
@hyginn I need to be sure that node that disallowing swapping between bridges is the right way to prevent disconnected components, before I can move on. Unfortunately igraph does not have such a function yet.